patroni/scripts/barman/config_switch.py

```diff
@@ -0,0 +1,155 @@
+        ret = True
+    elif args.switch_when == "demoted" and args.role not in ["replica",
+                                                             "demoted"]:
+        ret = True
+
+    return ret
+
+
+def _switch_config(api: "PgBackupApi", barman_server: str,
+                   barman_model: Optional[str], reset: Optional[bool]) -> bool:
+    """Switch configuration of Barman server through ``pg-backup-api``.
+
+    .. note::
+        If requests to ``pg-backup-api`` fail recurrently or we face HTTP
+        errors, then exit with :attr:`ExitCode.HTTP_ERROR`.
+
+    :param api: a :class:`PgBackupApi` instance to handle communication with
+        the API.
+    :param barman_server: name of the Barman server which config is to be
+        switched.
+    :param barman_model: name of the Barman model to be applied to the server,
+        if any.
+    :param reset: ``True`` if you would like to unapply the currently active
+        model for the server, if any.
+
+    :returns: ``True`` if config was successfully switched, ``False``
+        otherwise.
+    """
+    operation_id = None
+
+    try:
+        operation_id = api.create_config_switch_operation(
+            barman_server,
+            barman_model,
+            reset,
+        )
+    except RetriesExceeded as exc:
+        logging.error("An issue was faced while trying to create a config "
+                      "switch operation: %r", exc)
+        sys.exit(ExitCode.HTTP_ERROR)
+
+    logging.info("Created the config switch operation with ID %s",
+                 operation_id)
+
+    status = None
+
+    while True:
+        try:
+            status = api.get_operation_status(barman_server, operation_id)
+        except RetriesExceeded:
+            logging.error("Maximum number of retries exceeded, exiting.")
+            sys.exit(ExitCode.HTTP_ERROR)
+
+        if status != OperationStatus.IN_PROGRESS:
+            break
+
+        logging.info("Config switch operation %s is still in progress",
+                     operation_id)
+        time.sleep(5)
+
+    return status == OperationStatus.DONE
+
+
+def run_barman_config_switch(api: "PgBackupApi", args: Namespace) -> None:
+    """Run a remote ``barman config-switch`` through the ``pg-backup-api``.
+
+    :param api: a :class:`PgBackupApi` instance to handle communication with
+        the API.
+    :param args: arguments received from the command-line of
+        ``patroni_barman config-switch`` command.
+    """
+    if _should_skip_switch(args):
+        logging.info("Config switch operation was skipped (role=%s, "
+                     "switch_when=%s).", args.role, args.switch_when)
+        sys.exit(ExitCode.CONFIG_SWITCH_SKIPPED)
+
+    barman_model, reset = args.barman_model, args.reset
+
+    if all([barman_model, reset]) or not any([barman_model, reset]):
+        logging.error("One, and only one among 'barman_model' ('%s') and "
+                      "'reset' ('%s') should be given", barman_model, reset)
```

Comment:
I think we could use XOR here :)
```suggestion
    if not bool(args.barman_model) ^ bool(args.reset):
        logging.error("One, and only one among 'barman_model' ('%s') and "
                      "'reset' ('%s') should be given", args.barman_model, args.reset)
```

Reply:
I like this -- didn't thought of that, much cleaner :)